src/lib/draw-and-pick.js

```diff
@@ -212,6 +263,93 @@ function pickFromBuffer(gl, {
   const width = Math.min(pickingFBO.width, deviceX + deviceRadius) - x + 1;
   const height = Math.min(pickingFBO.height, deviceY + deviceRadius) - y + 1;
 
+  const pickedColors = getPickedColors(gl, {layers, pickingFBO, deviceRect: {x, y, width, height}});
+
+  // Traverse all pixels in picking results and find the one closest to the supplied
+  // [deviceX, deviceY]
+  let minSquareDistanceToCenter = deviceRadius * deviceRadius;
+  let closestResultToCenter = null;
+  let i = 0;
+
+  for (let row = 0; row < height; row++) {
+    for (let col = 0; col < width; col++) {
+      // Decode picked layer from color
+      const pickedLayerIndex = pickedColors[i + 3] - 1;
+
+      if (pickedLayerIndex >= 0) {
+        const dx = col + x - deviceX;
+        const dy = row + y - deviceY;
+        const d2 = dx * dx + dy * dy;
+
+        if (d2 <= minSquareDistanceToCenter) {
+          minSquareDistanceToCenter = d2;
+
+          // Decode picked object index from color
+          const pickedColor = pickedColors.slice(i, i + 4);
+          const pickedLayer = layers[pickedLayerIndex];
+          const pickedObjectIndex = pickedLayer.decodePickingColor(pickedColor);
+          closestResultToCenter = {pickedColor, pickedLayer, pickedObjectIndex};
+        }
+      }
+      i += 4;
+    }
+  }
+
+  return closestResultToCenter || {
+    pickedColor: EMPTY_PIXEL,
+    pickedLayer: null,
+    pickedObjectIndex: -1
+  };
+}
+
+/**
+ * Query within a specified rectangle
+ * Returns array of unique objects in shape `{x, y, pickedColor, pickedLayer, pickedObjectIndex}`
+ */
+function queryFromBuffer(gl, {
```

Comment:
I'm really bad at naming stuff... but we might consider change the name of queryFromBuffer() as framebuffer is not buffer... people with webgl knowledge will try to find which buffer this function executes on and figure out there is no buffer involved.

I propose queryFromPickingBuffer? or queryForRect